### DISCUSSION TITLE
How can I create compound variant sortable api on top dnd-kit?

### DISCUSSION THREAD

**Discussion by joseDaKing** [clauderic/dnd-kit#680] [Q&A] [answered]
How can I create compound variant sortable API on top dnd-kit?

**Comment by joseDaKing** [comment]
It seems unnecessary to do that

**Comment by clauderic** [comment, marked as answer]
I'm not sure why you would want to do that considering there's a built-in sortable preset (`@dnd-kit/sortable`).

Refer to https://docs.dndkit.com/presets/sortable

**Reply by joseDaKing** [reply]
@clauderic 

The reason why is that a don't have to convert some hierarchal objects to lists and a list back to a hierarchal object, it would become somewhat tedious but still think it is still unnecessary to have or write a compound variant API. For example, I have an object that looks like this
```ts
type Items = {
  items: string[];
};

type State = {
  // section 1 - 3 are required section where the order can be changed
  section1: Items;
  section2: Items;
  section3: Items;

  // section 4 - 6 are optional sections that can be added and can also be changed
  section4?: Items;
  section5?: Items;
  section6?: Items;
}
```
